plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/functions/tablechanges/TableChangesFunction.java

```diff
@@ -0,0 +1,180 @@
+    private static final String COMMIT_VERSION_COLUMN_NAME = "_commit_version";
+    private static final String COMMIT_TIMESTAMP_COLUMN_NAME = "_commit_timestamp";
+    private static final Set<String> IGNORED_COLUMNS = ImmutableSet.of(PATH_COLUMN_NAME, FILE_SIZE_COLUMN_NAME, FILE_MODIFIED_TIME_COLUMN_NAME);
+
+    private final TrinoFileSystemFactory fileSystemFactory;
+    private final DeltaLakeMetadataFactory deltaLakeMetadataFactory;
+
+    public TableChangesFunction(TrinoFileSystemFactory fileSystemFactory, DeltaLakeMetadataFactory deltaLakeMetadataFactory)
+    {
+        super(
+                SCHEMA_NAME,
+                NAME,
+                ImmutableList.of(
+                        ScalarArgumentSpecification.builder().name(SCHEMA_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(TABLE_NAME_ARGUMENT).type(VARCHAR).build(),
+                        ScalarArgumentSpecification.builder().name(SINCE_VERSION_ARGUMENT).type(BIGINT).defaultValue(-1L).build()),
+                GENERIC_TABLE);
+        this.fileSystemFactory = requireNonNull(fileSystemFactory, "fileSystemFactory is null");
+        this.deltaLakeMetadataFactory = requireNonNull(deltaLakeMetadataFactory, "deltaLakeMetadataFactory is null");
+    }
+
+    @Override
+    public TableFunctionAnalysis analyze(ConnectorSession session, ConnectorTransactionHandle transaction, Map<String, Argument> arguments)
+    {
+        ScalarArgument schemaNameArgument = (ScalarArgument) arguments.get(SCHEMA_NAME_ARGUMENT);
+        checkFunctionArgument(schemaNameArgument.getValue() != null, "schema_name value for function table_changes() cannot be null");
+        String schemaName = ((Slice) schemaNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument tableNameArgument = (ScalarArgument) arguments.get(TABLE_NAME_ARGUMENT);
+        checkFunctionArgument(tableNameArgument.getValue() != null, "table_name value for function table_changes() cannot be null");
+        String tableName = ((Slice) tableNameArgument.getValue()).toStringUtf8();
+
+        ScalarArgument sinceVersionArgument = (ScalarArgument) arguments.get(SINCE_VERSION_ARGUMENT);
+        if (sinceVersionArgument.getValue() == null) {
+            throw new TrinoException(INVALID_FUNCTION_ARGUMENT, "since_version argument must be a number, null provided");
+        }
+        long sinceVersion = (long) sinceVersionArgument.getValue() + 1; //+1 to ensure that the sinceVersion is exclusive
+        checkFunctionArgument(sinceVersion >= 0, "since_version value for function table_changes() cannot be negative");
+
+        DeltaLakeMetadata deltaLakeMetadata = deltaLakeMetadataFactory.create(session.getIdentity());
+        SchemaTableName schemaTableName = new SchemaTableName(schemaName, tableName);
+        DeltaLakeTableHandle tableHandle = (DeltaLakeTableHandle) deltaLakeMetadata.getTableHandle(session, schemaTableName);
```

Comment:
I was trying the functionality of `table_changes` on a corrupted table and stumbled on this kind of error:

```
trino:default> select * from TABLE(delta.system.table_changes('default', 'cdfmariustest250', 21));
Query 20230403_144154_00000_skbrv failed: class io.trino.plugin.deltalake.CorruptedDeltaLakeTableHandle cannot be cast to class io.trino.plugin.deltalake.DeltaLakeTableHandle (io.trino.plugin.deltalake.CorruptedDeltaLakeTableHandle and io.trino.plugin.deltalake.DeltaLakeTableHandle are in unnamed module of loader io.trino.server.PluginClassLoader @1690305e)
java.lang.ClassCastException: class io.trino.plugin.deltalake.CorruptedDeltaLakeTableHandle cannot be cast to class io.trino.plugin.deltalake.DeltaLakeTableHandle (io.trino.plugin.deltalake.CorruptedDeltaLakeTableHandle and io.trino.plugin.deltalake.DeltaLakeTableHandle are in unnamed module of loader io.trino.server.PluginClassLoader @1690305e)
	at io.trino.plugin.deltalake.functions.tablechanges.TableChangesFunction.analyze(TableChangesFunction.java:113)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorTableFunction.analyze(ClassLoaderSafeConnectorTableFunction.java:78)
	at io.trino.sql.analyzer.StatementAnalyzer$Visitor.visitTableFunctionInvocation(StatementAnalyzer.java:1572)
```

cc @findepi  maybe this kind of cast breaks other methods in `DeltaLakeMetadata` as well.

Reply:
This would probably break any code that expect different type of TableHandle? 
But that is quite clear programmer error so I doubt I should do anything about it

Reply:
Context 

https://github.com/trinodb/trino/blob/e7d560ff347e242215706171b17590befcaa5a3d/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java#L419-L424


Add a corresponding test to the method:

https://github.com/trinodb/trino/blob/e7d560ff347e242215706171b17590befcaa5a3d/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java#L143